docker/jenkins/publish-build.ps1

```diff
@@ -99,9 +99,41 @@ $headers.Add("Accept", "application/vnd.github.v3+json")
 $headers.Add("Authorization", "token $pat")
 
 $url = "https://api.github.com/repos/rstudio/latest-builds/contents/content/rstudio/$flower/$build/$versionStem.md"
-
 # Send to Github! We have to use basic parsing here because this script runs on SKU of Windows that
 # doesn't contain a working copy of IE (and, incredibly, without -UseBasicParsing, Invoke-WebRequest
 # has a dendency on the IE DOM engine).
-Invoke-WebRequest -Body $payload -Method 'PUT' -Headers $headers -Uri $url -UseBasicParsing
-
+try
+{
+    $createResponse = Invoke-RestMethod -Body $payload -Method 'PUT' -Headers $headers -Uri $url -UseBasicParsing
+    Write-Host "Response :"
+    Write-Host $createResponse.Content
+} catch {
+    $StatusCode = $_.Exception.Response.StatusCode.value__
+    
+    if ($StatusCode -eq 422) # Assume the file already exists and we need to do an update
+    {
+        Write-Host "Received a 422 error, assuming it's an issue updating. Getting existing file's SHA"
+        $getSha = Invoke-RestMethod -Method 'GET' -Headers $headers -Uri $url -UseBasicParsing
+        Write-Host "Github Response:"
+        Write-Host $getSha
+        $updateSha = $getSha.sha
```

Comment:
Wouldn't this be a no-op? We get the sha that the dailies page already has and give it back to the dailies page?

Reply:
We need to retrieve the sha of the file in the dailies page repo to perform an update instead of a create.

Previously we were supplying a the binary sha as the github file sha and in a create, that didn't cause an issue, it was ignored. In the update case the github file sha didn't match the binary sha, producing the confusing error messages.

What this code is doing is getting the github file sha, so we can resubmit our change as an update, instead of as a create. The github api is using the file sha as a way of making sure a commit doesn't interleave and we don't end up stomping changes. See https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents